### PR TITLE
Add missing IP ranges to IsRoutable

### PR DESCRIPTION
**Pull request by quantumagi**
See story 2938.

https://dev.azure.com/Stratisplatformuk/StratisBitcoinFullNode/_workitems/edit/2938/

### CODE REVIEW

File: src/NBitcoin/IpExtensions.cs [hunk 1]
```diff
@@ -208,20 +281,26 @@ public static bool IsMulticast(this IPAddress address)
...
         /// </summary>
         public static bool IsRoutable(this IPAddress address, bool allowLocal)
         {
             return address.IsValid() && !(
                                             (!allowLocal && address.IsRFC1918()) ||
+                                            address.IsRFC5737() ||
+                                            address.IsRFC6598() ||
+                                            address.IsRFC7534() ||
+                                            address.IsRFC2544() ||
+                                            address.IsRFC1112() ||
```
Comment (fassadlr) [inline src/NBitcoin/IpExtensions.cs]: As RFC1112 is 240.0.0.0/4, should this be in this list? (Unless the comment is out of date)

Comment (quantumagi) [reply src/NBitcoin/IpExtensions.cs]: The reasoning is that seeing that RFC 1112 is reserved it may not be routable.

Comment (fassadlr) [reply src/NBitcoin/IpExtensions.cs]: Ah my bad @quantumagi! You are 100% correct. I didnt see the `!`.

**Comment by fassadlr** [comment]
@maciejzaleski Can you double check this as well please?

File: src/NBitcoin/IpExtensions.cs [hunk 2]
```diff
@@ -7,6 +7,80 @@ namespace NBitcoin
...
+                (bytes[15 - 3] == 203 && bytes[15 - 2] == 0 && bytes[15 - 1] == 113));
+        }
+
+        /// <summary>100.64.0.0/10 Shared Address Space. Not globally reachable.</summary>
+        public static bool IsRFC6598(this IPAddress address)
+        {
+            address = address.EnsureIPv6();
+            byte[] bytes = address.GetAddressBytes();
+            return address.IsIPv4() && (
+                bytes[15 - 3] == 100 && (bytes[15 - 2] & 0xc0) == 64);
```
Comment (noescape00) [inline src/NBitcoin/IpExtensions.cs]: Why `& 0xc0` is here?

File: src/NBitcoin/IpExtensions.cs [hunk 3]
```diff
@@ -7,6 +7,80 @@ namespace NBitcoin
...
+        public static bool IsRFC7535(this IPAddress address)
+        {
+            address = address.EnsureIPv6();
+            byte[] bytes = address.GetAddressBytes();
+            return address.IsIPv4() && (
+                bytes[15 - 3] == 192 && bytes[15 - 2]  == 31 && bytes[15 - 1] == 196);
+        }
+
+        /// <summary>192.52.193.0/24 AMT. Globally reachable.</summary>
+        public static bool IsRFC7450(this IPAddress address)
```
Comment (noescape00) [inline src/NBitcoin/IpExtensions.cs]: This and the next method are not used

**Comment by fassadlr** [comment]
Fixes #2888

**Pull request merged**